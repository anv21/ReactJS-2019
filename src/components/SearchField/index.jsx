import React from 'react';
import PropTypes from 'prop-types';

import SEARCH_BY from "../../constants/SEARCH_BY";

const ENTER_KEY = 13;

class SearchField extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            value: this.props.value
        }
    }

    onChange = (event) => {
        this.setState({
            value: event.target.value
        })
    }

    onEnter = (event) => {
        const isEnterPressed = event.which === ENTER_KEY || event.keyCode === ENTER_KEY;
        if (isEnterPressed) {
            this.props.setSearchValue(this.state.value);
            this.props.getItems();
        }
    }

    onSearchByTitleClick = () => {
        this.props.setSearchBy(SEARCH_BY.TITLE);
    }

    onSearchByGenreClick = () => {
        this.props.setSearchBy(SEARCH_BY.GENRE);
    }

    onSearchClick = () => {
        this.props.getItems();
    }

    onBlur = () => {
        this.props.setSearchValue(this.state.value);
    }

    render() {
        const {searchBy} = this.props;

        return (
            <div className="search_field_wrapper">
                <label className="search_field_label">find your movie</label>
                <input className="search_field_input"
                    onChange={this.onChange}
                    onBlur={this.onBlur}
                    onKeyPress={this.onEnter}
                    type="text"
                    value={this.state.value}
                />
                <div>
                    <label className="search_field_label">search by</label>
                    <button
                        className={"search_by_button " + (searchBy === SEARCH_BY.TITLE && "active_button")}
                        onClick={this.onSearchByTitleClick}
                    >title
                    </button>
                    <button
                        className={"search_by_button " + (searchBy === SEARCH_BY.GENRE && "active_button")}
                        onClick={this.onSearchByGenreClick}
                    >genre
                    </button>
                    <button
                        className="search_button"
                        type="button"
                        onClick={this.onSearchClick}
                    >search
                    </button>
                </div>
            </div>
        )
    }
}

SearchField.propTypes = {
    value: PropTypes.string,
    searchBy: PropTypes.string,
    getItems: PropTypes.func,
    setSearchBy: PropTypes.func,
};

SearchField.defaultProps = {
    value: '',
    searchBy: SEARCH_BY.TITLE,
    getItems: null,
    setSearchBy: null,
};

export default SearchField;