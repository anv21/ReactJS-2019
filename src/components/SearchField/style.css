.search_field_wrapper {
    padding: 40px 0;
}

.search_field_label, 
.search_field_input,
.search_by_button,
.search_button {
    color: #fff;
}

.search_field_label {
    text-transform: uppercase;
}

.search_field_input {
    width: 100%;
    margin: 10px 0;
    padding: 20px 25px;
    background-color: rgba(0, 0, 0, 0.9);
    border: none;
    border-bottom: 2px solid #f55263;
    font-size: 16px;
    box-sizing: border-box;
    outline: none;
}

.search_by_button,
.search_button {
    cursor: pointer;
    border: none;
    border-radius: 5px;
    text-transform: uppercase;
}

.search_by_button {
    background-color: #3c3c3c;
    padding: 5px 20px;
    margin: 20px 0 0 10px;
}

.search_by_button:hover {
    background-color: #f55263;
}

.active_button {
    background-color: #f55263;
}

.search_button {
    float: right;
    font-weight: bold;
    background-color: #f55263;
    padding: 10px 60px;
    margin-top: 20px;
}

.search_button:hover {
    background-color: #fff;
    color: #f55263;
}