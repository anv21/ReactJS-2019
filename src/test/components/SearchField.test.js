import React from 'react';
import { shallow } from 'enzyme';

import SearchField from '../../components/SearchField/index.jsx';

describe('SearchField', () => {
    const mockGetItems = jest.fn();
    const searchField = shallow(<SearchField getItems={mockGetItems} />);

    it('should be like snapshot', () => {
        expect(searchField).toMatchSnapshot();
    });

    it('should call getItems function on SearchButton click', () => {
        searchField.find('.search_button').simulate('click');
        expect(mockGetItems.mock.calls.length).toBe(1);
    });
});
