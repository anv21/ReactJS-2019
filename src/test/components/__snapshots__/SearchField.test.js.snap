// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField should be like snapshot 1`] = `ShallowWrapper {}`;
